### PR TITLE
Optimize jump bar

### PR DESCRIPTION
**Pull request by romgrk**
Replaces fzy python scoring with a PG fzy extension.

Related but not dependent on: https://github.com/c3g/freezeman_web/pull/46

Install https://github.com/romgrk/pg_fzy, then run the migration file or `create extension pg_fzy;` in the database.

### CODE REVIEW

**Comment by jleblond** [comment]
It does a good job at improving the speed at which the results are returned. These might be just details, but I was wondering though if there is a way to improve its accuracy and responsiveness. It seems that the order of the results based on the query similarities is not quite accurate too. Should we consider exact start matching as being of higher similarity than having the string contained at the end of the query? Also, do we need to see other results which are different from the query we typed?  Last point, you mentioned this is working better for quick typers, but if you are too quick (as in 'you know exactly the name you are looking for'), you might get confused by the results being displayed to you due to the delay.

as an example:
image

**Comment by romgrk** [comment]
Observations, `score` that now comes from the postgres trigram similarity() function doesn't seem very correct:

Screenshot from 2021-02-25 09-15-47
Screenshot from 2021-02-25 09-16-07

**Comment by romgrk** [comment]
PG Extension: https://github.com/romgrk/pg_fzy

File: fms_core/viewsets.py [hunk 1]
```diff
@@ -3,7 +3,9 @@
 from collections import Counter
 from django.conf import settings
 from django.contrib.auth.models import User, Group
-from django.db.models import Count, Q
+from django.contrib.postgres.search import TrigramSimilarity
```
Comment (UlysseFG) [inline fms_core/viewsets.py:6]: Might want to drop this import of TrigramSimilarity

**Comment by romgrk** [comment]
Need to make sure full matches are returned (first) by the matching function.

**Review by UlysseFG — COMMENTED** [review]
Don't you need to add the binaries somewhere in the repo ? I don't really know the way these extensions work but i do not know where this links to your fzy.

**Comment by romgrk** [comment]
The link is in the top-comment but I'll add the extension as a submodule & an install script.

**Comment by romgrk** [comment]
I've added the extension as a git submodule and all the details on the readme. Can you read it and let me know if it's clear enough? As is mentioned, you'll need to initialize the new submodule (`git submodule update --init --recursive`).

File: README.md [hunk 2]
```diff
@@ -160,3 +169,16 @@ coverage run ./manage.py test
...
+The `pg_fzy` extension is a PostgreSQL C extension that implements the fzy
+algorithm. To manage it, here are the useful commands, to be run in the
+`./dependencies/pg_fzy` directory:
+
+ - `make`: build the shared library (linux-x86_64 included by default)
+ - `sudo make install`: install it in the postgres extension list
+ - `sudo make uninstall`: remove it from the postgres extension list
+
+When updating the extension, you might need to run `drop extension pg_fzy;` and
+`create extension fzy;` before uninstalling and after installing it.
```
[the remark this answers is not in the document]
Comment (romgrk) [reply README.md:184]: yes

[the remark this answers is not in the document]
Comment (romgrk) [reply README.md:184]: No, I just read the typo, it's `fzy` in both cases

**Pull request merged**